tt_eager/tt_dnn/op_library/rotary_embedding/rotary_embedding_llama_op.cpp

```diff
@@ -0,0 +1,112 @@
+    TT_FATAL(trans_mat.get_legacy_shape()[-2] == TILE_HEIGHT, "Transformation matrix must have 3rd dim equal to TILE_HEIGHT");
+    TT_FATAL(trans_mat.get_legacy_shape()[-1] == TILE_WIDTH, "Transformation matrix must have 4rd dim equal to TILE_WIDTH");
+
+
+    TT_FATAL(input_tensor.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED);
+    TT_FATAL(this->output_mem_config.memory_layout == TensorMemoryLayout::INTERLEAVED);
+
+}
+
+std::vector<Shape> RotaryEmbeddingLlama::compute_output_shapes(const std::vector<Tensor>& input_tensors) const {
+    const auto& input_tensor = input_tensors.at(0);
+    auto shape = input_tensor.get_legacy_shape();
+    return {shape};
+}
+
+std::vector<Tensor> RotaryEmbeddingLlama::create_output_tensors(const std::vector<Tensor>& input_tensors) const {
+    const auto& input_tensor = input_tensors.at(0);
+    auto output_shape = this->compute_output_shapes(input_tensors)[0];
+    return {create_device_tensor(
+        output_shape, input_tensor.get_dtype(), input_tensor.get_layout(), input_tensor.device(), this->output_mem_config)};
+}
+
+operation::ProgramWithCallbacks RotaryEmbeddingLlama::create_program(
+    const std::vector<Tensor>& input_tensors, std::vector<Tensor>& output_tensors) const {
+    const auto& input_tensor = input_tensors.at(0);
+    const auto& cos = input_tensors.at(1);
+    const auto& sin = input_tensors.at(2);
+    const auto& trans_mat = input_tensors.at(3);
+    auto& output_tensor = output_tensors.at(0);
+
+    switch (this->get_parallelization_strategy(input_tensors)) {
+        case RotaryEmbeddingLlamaOpParallelizationStrategy::MULTI_CORE:
+            return rotary_embedding_llama_multi_core(input_tensor, cos, sin, trans_mat, output_tensor, this->compute_kernel_config);
+            break;
+        case RotaryEmbeddingLlamaOpParallelizationStrategy::SINGLE_CORE:
+        default: return rotary_embedding_llama_single_core(input_tensor, cos, sin, trans_mat, output_tensor, this->compute_kernel_config);
+    }
+}
+
+RotaryEmbeddingLlamaOpParallelizationStrategy RotaryEmbeddingLlama::get_parallelization_strategy(
+    const std::vector<Tensor>& input_tensors) const {
+    const auto& input_tensor = input_tensors.at(0);
+    // num_rows = 1 x 8 x 128 x 128 / 128 / 32 = 32
+    uint32_t num_rows = input_tensor.volume() / input_tensor.get_legacy_shape()[-1] / TILE_HEIGHT;
+    if (num_rows > 1) {
+        return RotaryEmbeddingLlamaOpParallelizationStrategy::MULTI_CORE;
+    } else {return RotaryEmbeddingLlamaOpParallelizationStrategy::SINGLE_CORE;}
+    return RotaryEmbeddingLlamaOpParallelizationStrategy::SINGLE_CORE;
+}
+
+tt::stl::reflection::Attributes RotaryEmbeddingLlama::attributes() const {
+    return {
+        {"seq_len", this->seq_len},
+        {"output_mem_config", this->output_mem_config},
+    };
+}
+
+const operation::Hash RotaryEmbeddingLlama::compute_program_hash(const std::vector<Tensor>& input_tensors) const {
```

Comment:
We shouldn't necessarily implement `compute_program_hash` here. Looks like this implements default hashing anyways, so please remove.